pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,288 @@
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateTrackingOptions checks if TrackingOptions Object are up to date
+func isUpToDateTrackingOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.TrackingOptions != nil && resp.TrackingOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.TrackingOptions.CustomRedirectDomain) != awsclient.StringValue(resp.TrackingOptions.CustomRedirectDomain) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateSuppressionOptions checks if SuppressionOptions Object are up to date
+func isUpToDateSuppressionOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	crSuppressedReasons := make([]*string, 0)
+	awsSuppressedReasons := make([]*string, 0)
+
+	if cr.Spec.ForProvider.SuppressionOptions != nil && cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons != nil {
+		crSuppressedReasons = append(crSuppressedReasons, cr.Spec.ForProvider.SuppressionOptions.SuppressedReasons...)
+	}
+
+	if resp.SuppressionOptions != nil && resp.SuppressionOptions.SuppressedReasons != nil {
+		awsSuppressedReasons = resp.SuppressionOptions.SuppressedReasons
+	}
+
+	if len(crSuppressedReasons) != len(awsSuppressedReasons) {
+		return false
+	}
+
+	sortCmp := cmpopts.SortSlices(func(i, j *string) bool {
+		return aws.StringValue(i) < aws.StringValue(j)
+	})
+
+	return cmp.Equal(crSuppressedReasons, awsSuppressedReasons, sortCmp)
+
+}
+
+// isUpToDateSendingOptions checks if SendingOptions Object are up to date
+func isUpToDateSendingOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	if cr.Spec.ForProvider.SendingOptions != nil && resp.SendingOptions != nil {
+		if awsclient.BoolValue(cr.Spec.ForProvider.SendingOptions.SendingEnabled) != awsclient.BoolValue(resp.SendingOptions.SendingEnabled) {
+			return false
+		}
+	}
+	return true
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	switch awsclient.BoolValue(resp.SendingOptions.SendingEnabled) {
+	case true:
+		cr.Status.SetConditions(xpv1.Available())
+	case false:
+		cr.Status.SetConditions(xpv1.Unavailable())
+	default:
+		cr.Status.SetConditions(xpv1.Creating())
+	}
+
+	return obs, nil
+}
+
+func (e *hooks) update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) { // nolint:gocyclo
+	cr, ok := mg.(*svcapitypes.ConfigurationSet)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+	// Update Resource is not provided other than individual PUT operation
+	// NOTE: Update operation NOT generated by ACK code-generator
+
+	// update DeliveryOptions (PutConfigurationSetDeliveryOptions)
```

Comment:
Would it be possible for us to issue a `GET` call here and see what we need to update exactly so that we make less calls?

Reply:
Can you show me the concrete example on how i can achieve this?

Is there a way we can achieve via passing info from `IsUpToDate` as comparison of both cr and actual resource state is happening here?

Reply:
You can add fields to the `hooks` struct, set them during `Observe` and access them again during `Update` or `Create`.

Reply:
Will have a look on that shortly. If i understand correctly, there is only `preObserve` and `postObserve` func that can be overridden in the controller and for this case, `postObserve` would be a fit

Reply:
Have enhanced the update function to trigger only based on specific attribute changes